OPRUN-4056: [OTE] Migrate preflight checks from openshift/origin

camilamacedo86 opened the pull request:
Migrated OLMv1 operator preflight checks from using external YAML files to defining ClusterRole permissions directly in Go structs. This improves test reliability and simplifies test setup by removing file dependencies.

The changes ensure precise replication of original test scenarios, including specific permission omissions for services, create verbs, ClusterRoleBindings, ConfigMap resourceNames, and escalate/bind verbs.

You can check the PR to delete the migrated tests here : https://github.com/openshift/origin/pull/30074 ( it might help you validate )

Tests:

> https://pr-payload-tests.ci.openshift.org/runs/ci/cab853e0-7a77-11f0-80dc-0a55ab730901-0

<img width="875" height="139" alt="Screenshot 2025-08-18 at 06 53 34" src="https://github.com/user-attachments/assets/943fb494-446e-4792-8192-46ca4dde25a5" />

camilamacedo86 commented on openshift/tests-extension/.openshift-tests-extension/openshift_payload_olmv1.json:
```diff
@@ -1,4 +1,64 @@
+    "environmentSelector": {}
+  },
+  {
+    "name": "[sig-olmv1][OCPFeatureGate:NewOLMPreflightPermissionChecks][Skipped:Disconnected] OLMv1 operator preflight checks should report error when {create} verb is not specified",
```
Same name: https://github.com/openshift/origin/blob/7ccc307375be090b9a6fb44b713dcbfe318430be/test/extended/util/annotate/generated/zz_generated.annotations.go#L1988

camilamacedo86 commented on openshift/tests-extension/.openshift-tests-extension/openshift_payload_olmv1.json:
```diff
@@ -1,4 +1,64 @@
+    "environmentSelector": {}
+  },
+  {
+    "name": "[sig-olmv1][OCPFeatureGate:NewOLMPreflightPermissionChecks][Skipped:Disconnected] OLMv1 operator preflight checks should report error when {ConfigMap:resourceNames} are not all specified",
```
Same name: https://github.com/openshift/origin/blob/7ccc307375be090b9a6fb44b713dcbfe318430be/test/extended/util/annotate/generated/zz_generated.annotations.go#L1984

camilamacedo86 commented on openshift/tests-extension/.openshift-tests-extension/openshift_payload_olmv1.json:
```diff
@@ -1,4 +1,64 @@
+    "environmentSelector": {}
+  },
+  {
+    "name": "[sig-olmv1][OCPFeatureGate:NewOLMPreflightPermissionChecks][Skipped:Disconnected] OLMv1 operator preflight checks should report error when {clusterextension/finalizer} is not specified",
```
Same name: https://github.com/openshift/origin/blob/7ccc307375be090b9a6fb44b713dcbfe318430be/test/extended/util/annotate/generated/zz_generated.annotations.go#L1986

camilamacedo86 commented:
/retest-required

camilamacedo86 commented:
/test openshift-e2e-aws

tmshort commented:
/test openshift-e2e-aws-techpreview
Why are the tech preview tests failing?

kuiwang02 commented:
/lgtm

please note: the merge is held because of cutting branch. so, better to wait for that.

camilamacedo86 commented:
Hi @kuiwang02

This one, I think, we need to get merged.
It will help us to bump the 4.20 images ( which block the release ) 
With it, we also need: https://github.com/openshift/origin/pull/30084

So, we can bump and the PR pass on the tests. 

/lgtm

camilamacedo86 commented:
/retest-required

tmshort commented on openshift/tests-extension/test/olmv1-preflight.go:
I really think we should have `scenario` has an enum. Right now we have magic numbers 0 through 5.

tmshort commented:
Again... why are techpreview tests failing? I've seen this elsewhere... so I'm not sure it's related to _this_ PR.

camilamacedo86 commented:
/hold cancel

camilamacedo86 commented:
/test openshift-e2e-aws-techpreview
/test okd-scos-e2e-aws-ovn

camilamacedo86 commented:
/payload-aggregate-with-prs periodic-ci-openshift-release-master-nightly-4.20-e2e-metal-ipi-ovn-bm

camilamacedo86 commented:
/payload-aggregate periodic-ci-openshift-release-master-ci-4.20-e2e-gcp-ovn-techpreview-serial 10

camilamacedo86 commented:
/test openshift-e2e-aws

perdasilva commented on openshift/tests-extension/Makefile:
```diff
@@ -19,6 +19,7 @@ LDFLAGS := -X '$(GO_PKG_NAME)/pkg/version.CommitFromGit=$(GIT_COMMIT)' \
 
 METADATA := $(shell pwd)/.openshift-tests-extension/openshift_payload_olmv1.json
 
+
```
Do we need the changes in this file?

camilamacedo86 replied:
Fixed !!! 👍

camilamacedo86 commented:
Hi @perdasilva 

I rebased with main and addressed your comment ( remove unecessary space in the file which end up due rebases )
But we lose the lgtm.
Could you re-add both for we get it merged?

camilamacedo86 commented:
/payload-aggregate /payload-aggregate periodic-ci-openshift-release-master-ci-4.20-e2e-gcp-ovn-techpreview 5

camilamacedo86 commented:
/hold 

Until we get the result of the latest

See details on https://pr-payload-tests.ci.openshift.org/runs/ci/2adbf990-7db4-11f0-8f39-b7de71564e96-0

Just to be very extra safe
